edgedelta/templates/agent_updater.yaml

```
@@ -29,10 +29,10 @@ roleRef:
   name: agent-updater-roles
   apiGroup: rbac.authorization.k8s.io
 ---
-{{- if .Capabilities.APIVersions.Has "batch/v1beta1" }}
-apiVersion: batch/v1beta1
-{{- else }}
+{{- if .Capabilities.APIVersions.Has "batch/v1" }}
 apiVersion: batch/v1
+{{- else }}
+apiVersion: batch/v1beta1
```

This version would be better as it applies 1-1 correspondence between the capability check and the template.
```suggestion
{{- if .Capabilities.APIVersions.Has "batch/v1" }}
apiVersion: batch/v1
{{- else if .Capabilities.APIVersions.Has "batch/v1beta1"  }}
apiVersion: batch/v1beta1
{{- else }}
{{ fail "Neither 'batch/v1' nor 'batch/v1beta1' is supported by the cluster"
```